src/GafferSceneModule/HierarchyBinding.cpp

```diff
@@ -162,6 +162,18 @@ void GafferSceneModule::bindHierarchy()
 			.attr( "__qualname__" ) = "Instancer.ContextVariablePlug"
 		;
 
+		// Expose InstancerCapsules as if they were plain Capsules. We don't
+		// want to bind them fully because then we'd be exposing a private class, but
+		// we need to register them so that they can be returned to Python
+		// successfully when testing expansion in Python
+		//
+		// See "Boost.Python and slightly more tricky inheritance" at
+		// http://lists.boost.org/Archives/boost/2005/09/93017.php for more details.
+
+		boost::python::objects::copy_class_object(
+			type_id<Capsule>(), Instancer::instancerCapsuleTypeInfo()
+		);
```

Comment:
The "Selection Mask" menu in the viewer is broken now, always showing the highlighted "non default" icon, and refusing to tick the Capsules item. I suspect it's something to do with this, but haven't verified.

Reply:
I think I'll leave it to you to sort this one out, since you know the _SelectionMaskPlugValueWidget.

The cause of the break is pretty understandable. _SelectionMaskPlugValueWidget needs to find the types inheriting from a base class which are actually going to be rendered ... and it approximates this with _leafTypes, which returns any derived types which do not themselves have derived types. From a theoretical perspective, I think this is simply wrong ... just because something has a derived type doesn't mean it can't also be an object that gets rendered.

_SelectionMaskPlugValueWidget then compares sets of leaf types, but in the case of the hardcoded menu options, it compares leaf types to the hardcoded type ( rather than converting the hardcoded type to leaf types before comparing ). If this was fixed, then selecting the menu option would work ... but it would fail to actually mask Capsules that aren't InstancerCapsules.

So the trigger isn't the python binding stuff, but just the `IE_CORE_DECLAREEXTENSIONOBJECT( InstancerCapsule, GafferScene::InstancerCapsuleTypeId, GafferScene::Capsule );` which declares something deriving from Capsule ... I don't think there's anything wrong with this though.

What I think a fix would look like is:
* rather than _leafTypes, it should use something like _descendantTypes, which includes a type itself and all descendants whether or not they are leaves. This would include some types which actually abstract rather than concrete renderable things ... I wouldn't think that would actually screw it up, but I might be missing something
* rather than `checked = currentTypes.issuperset( types )` it feels like it might be more consistent to do `checked = currentTypes.issuperset( _descendantTypes( types ) )` so that both sides of the comparison have been processed through _descendantTypes ... though I guess switching from _leafTypes to _descendantTypes means this isn't strictly necessary

Anyway, I can do the work if this sounds like the right approach, but I figured I check in before trying to change the meanings of some stuff I don't fully understand.